src/bicep-registry/scheduled-action/main.bicep

```diff
@@ -0,0 +1,138 @@
+//   param scope string
+
+@description('Specifies which built-in view to use. This is a shortcut for the full view ID.')
+@allowed([
+  ''
+  'AccumulatedCosts'
+  'DailyCosts'
+  'InvoiceDetails'
+])
+param builtInView string = ''
+
+@description('Required if kind is "Email" and builtInView is not set. The resource ID of the view to which the scheduled action will send. The view must either be private (tenant level) or owned by the same scope as the scheduled action. Ignored if kind is "InsightAlert" or if builtInView is set.')
+param viewId string = ''
+
+@description('The display name to show in the portal when viewing the list of scheduled actions.')
+param displayName string = name
+
+@description('The status of the scheduled action.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param status string = 'Enabled'
+
+@description('Email address of the person or team responsible for this scheduled action. This email address will be included in emails.')
+param notificationEmail string = ''
+
+@description('List of email addresses that should receive emails. At least one valid email address is required.')
+param emailRecipients array
+
+@description('The subject of the email that will be sent to the email recipients.')
+@maxLength(70)
+param emailSubject string = ''
+
+@description('Include a message for recipients to add context about why they are getting the email, what to do, and/or who to contact.')
+@maxLength(250)
+param emailMessage string = ''
+
+@description('The language that will be used for the email template.')
+param emailLanguage string = 'en'
+
+@description('The regional format that will be used for dates, times, and numbers.')
+param emailRegionalFormat string = 'en-US'
+
+@description('Indicates whether to include a link to a CSV file with the backing data for the chart. Ignored if kind is "InsightAlert".')
+param includeCsv bool = false
+
+@description('The frequency at which the scheduled action will run.')
+@allowed([
+  'Daily'
+  'Weekly'
+  'Monthly'
+])
+param scheduleFrequency string = kind == 'InsightAlert' ? 'Daily' : 'Weekly'
+
+@description('Required if kind is "Email" and scheduleFrequency is "Weekly". List of days of the week that emails should be delivered. Allowed: Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday.')
+param scheduleDaysOfWeek array = [ 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday' ]
+
+@description('Required if kind is "Email" and scheduleFrequency is "Monthly". The day of the month that emails should be delivered. Note monthly cost is not final until the 3rd of the month. This or scheduleWeeksOfMonth is required if scheduleFrequency is "Monthly".')
+@allowed([ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31 ])
```

Comment:
1. Why does this start with 0?
2. API specifies Must be between 1 and 31

Reply:
0 = not set. I'll cover that in the description. It's the equivalent of '' in string parameters. We can't use null as an allowed value or default :-(